crates/sequencer/src/sequencer.rs

```diff
@@ -641,4 +665,91 @@ where
         rpc_methods.merge(rpc).unwrap();
         Ok(rpc_methods)
     }
+
+    #[allow(clippy::too_many_arguments)]
+    pub async fn await_commitment_tx_and_store(
+        &self,
+        tx_id: OneshotReceiver<Result<<Da as DaService>::TransactionId, <Da as DaService>::Error>>,
+        db_config: Option<SharedBackupDbConfig>,
+        l1_start_height: u64,
+        l1_end_height: u64,
+        commitment: SequencerCommitment,
+        l2_range: RangeInclusive<BatchNumber>,
+        commitment_info: CommitmentInfo,
+    ) {
+        // spawn an async task in the background and await the tx_id then save to pg
+        let ledger_db = self.ledger_db.clone();
+        tokio::spawn(async move {
+            match tx_id.await {
```

Comment:
In case sequencer is killed between sending commitment to DA and receiving it's tx, the commitment is lost anyway.

For example, here https://github.com/Sovereign-Labs/sovereign-sdk-wip/blob/fdeef09b345b377fed553a446df762880edd2123/full-node/sov-stf-runner/src/prover_service/manager/mod.rs there is an implementation of a queue of proofs to send to DA and retry if it doesn't work.

And here https://github.com/Sovereign-Labs/sovereign-sdk-wip/blob/fdeef09b345b377fed553a446df762880edd2123/full-node/sov-stf-runner/src/runner.rs#L289C12-L289C37 there is a code to stay in sync with DA.

So adding a permanent storage (sqlite or pg or whatever) would be an interesting choice.